mantidproject/mantid
# Run system tests with mantidpython

The system tests are now run with the `mantidpython` wrappers rather than through Vanilla Python. This ensures that the environment is set up correctly and should simulate the user experience more thoroughly.

In addition to this, a script is now generated in the _build_ directory that allows developers to more easily run the systemtests. It removes the requirement to know about the location of the runner scripts and the syntax follows that of `ctest`.

**To test:**

This will need to be tested on all platforms so can the main reviewer please coordinate with others when testing and have them annotate the PR.
- Run cmake and a file called `systemtest` or `systemtest.bat` will be created in the _build_ directory
- For _Linux_ or _OSX_ with _Makefiles/Ninja_ then the tests should now be able to be run using the following, e.g.

``` shell
cd build
./systemtest -R MARI
```
- For _Visual Studio_ or _Xcode_, the so-called multi-configuration generators, then the script requires the same `-C <cfg>` arguments that `ctest` requires and you must also build the `mantidpython` target first before running these commands

``` shell
cd build
systemtest -C Release -R MARI
```
- Check the scripts give sensible errors when not passed a `-C` argument.

Fixes #15395.

_Does not need to be in the release notes._

Tested platforms (add your name and date/time):
- [x] osx-10.9 @quantumsteve 
- [ ] rhel6
- [x] rhel7 @velynch 2016/4/1
- [x] ubuntu-14.04 @DanNixon 4/3
- [x] win7 - @FedeMPouzols tested debug/release, 2016/4/1

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
